cloud/amazon/ec2_asg_find.py

```diff
@@ -0,0 +1,410 @@
+                    {
+                        "key": "env",
+                        "propagate_at_launch": true,
+                        "resource_id": "public-webapp-production",
+                        "resource_type": "auto-scaling-group",
+                        "value": "production"
+                    }
+                ],
+                "termination_policies":
+                [
+                    "Default"
+                ],
+                "vpc_zone_identifier":
+                [
+                    "subnet-a1b1c1d1",
+                    "subnet-a2b2c2d2",
+                    "subnet-a3b3c3d3"
+                ]
+            }
+        ]
+    """
+    asgs = conn.describe_auto_scaling_groups()['AutoScalingGroups']
+    matched_asgs = []
+    for asg in asgs:
+        if name:
+            name_prog = re.compile(r'^' + name)
+            matched_name = name_prog.search(asg['AutoScalingGroupName'])
+        else: matched_name = True
+
+        if tags: matched_tags = match_asg_tags(tags, asg)
+        else: matched_tags = True
+
+        if matched_name and matched_tags:
+            asg_details = dict(created_time=asg['CreatedTime'].isoformat())
+            camel_prog = re.compile('(?!^)([A-Z]+)')
+
+            for key in ['AutoScalingGroupARN', 'AutoScalingGroupName', 'AvailabilityZones',
+                        'DefaultCooldown', 'DesiredCapacity', 'HealthCheckGracePeriod',
+                        'HealthCheckType', 'LaunchConfigurationName', 'LoadBalancerNames',
+                        'MaxSize', 'MinSize', 'NewInstancesProtectedFromScaleIn',
+                        'PlacementGroup' 'Status', 'TerminationPolicies']:
+                new_key = camel_prog.sub(r'_\1', key).lower()
+                if key in asg:
+                    asg_details[new_key] = asg[key]
+                else:
+                    asg_details[new_key] = None
+
+            for key in ['EnabledMetrics', 'Instances', 'SuspendedProcesses', 'Tags']:
+                if key in asg:
+                    new_key = camel_prog.sub(r'_\1', key).lower()
+                    asg_details[new_key] = []
+                    for item in asg[key]:
+                        asg_details[new_key].append(dict((camel_prog.sub(r'_\1', k).lower(), v) for k, v in item.iteritems()))
+
+            if 'VPCZoneIdentifier' in asg:
+                asg_details['vpc_zone_identifier'] = asg['VPCZoneIdentifier'].split(',')
+            else:
+                asg_details['vpc_zone_identifier'] = None
+
+            matched_asgs.append(asg_details)
+    return matched_asgs
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            name=dict(type='str'),
+            tags=dict(type='dict'),
+            limit_results=dict(required=False, type='int', default=0),
+            no_result_action = dict(
+                required=False, type='str', default='fail',
+                choices = ['success', 'fail']
```

Comment:
I agree with you @tombamford. Change the default choice from fail to success. The original idea behind this module, was to search for an ASG and if the results are empty or if the number of results is more than limit_results, than this should fail.